### ISSUE TITLE
Flags enums are not correctly inferred.

### ISSUE THREAD

**Issue by michaelstaib** [ChilliCream/hotchocolate#1020]
**Describe the bug**
If we infer a enum type form a flags enum like `HotChocolate.Types.EnumType<System.Reflection.TypeAttributes` we will run in to an error  that the value names cannot be correctly inferred since each value can have multiple states leading to names like:

`AUTOLAYOUT, ANSICLASS, CLASS, PUBLIC`

**To Reproduce**
```csharp
var schema = SchemaBuilder.New()
                .AddQueryType(c => c
                    .Name("Query")
                    .Field("foo")
                    .Type<StringType>()
                    .Resolver("bar"))
                .AddType<HotChocolate.Types.EnumType<System.Reflection.TypeAttributes>>()
                .Create();
```

**Expected behavior**
This still has to be defined. Do we infer those as a list of enums. Leading to things like `[[MyFlagsEnum]]`?

**Environment (please complete the following information):**
 - Version 10.0.1

**Comment by jonmill** [comment]
@michaelstaib - I have a project that uses an enum flag and want to expose that through the Graph...since this is still an open issue, what's the recommended way?

**Comment by michaelstaib** [comment]
yes ... there are a couple of issues actually.... a flags enum potentially would need to be exposed as a list of enum values since it can have multiple switches. There are other things in this regard. Like we need to detect it is a flags enum and then handle it accordingly. So, it is still an issue.

**Comment by jonmill** [comment]
Gotcha; so it sounds like the guidance right now is to add a converter to treat it (from the client perspective) as a numeric value, but the converter can convert it to the enum flags? 

Is there a list of work to do for this? We use flags for a bunch of things in my company's product, so I'd be interested in helping out 😃

**Comment by SpiritBob** [comment]
@michaelstaib are there plans on fixing this?

**Comment by PascalSenn** [comment]
Output is easy, input is tricky

**Comment by tobias-tengler** [comment]
Closing this, since support has been merged in #5186
